contrib/opencensus-ext-logging/opencensus/ext/logging/trace.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2017, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from opencensus.log import TraceLogger
+
+
+def trace_integration(tracer=None):
+    """Replace the global default logging class with `TraceLogger`.
+
+    Loggers created after the integration will produce `LogRecord`s
+    with extra traceId, spanId, and traceSampled attributes from the opencensus
+    context.
+    """
+    logging.setLoggerClass(TraceLogger)
```

Comment:
The whole integration is effectively just this line? I don't know that it's worth asking the user to install a separate package for this.

Reply:
I think eventually we might want to move the logger class into the extension (or consider switching to `setLogRecordFactory`).
The next step would be include formatters as what we've discussed in https://github.com/census-instrumentation/opencensus-python/issues/616#issuecomment-484992593.

Reply:
Regarding whether it's worth asking the user, I guess we don't need to ask the user:
1. Library developers will just use `opencensus`, with no extensions.
2. Application developers will pick at least one exporter, that exporter can include the logging package (e.g. I would imagine Azure customers to use `pip install opencensus-ext-azure`, which implies `opencensus` and `opencensus-ext-logging`).

Reply:
I see, I forgot about that comment. It looks like _I_ wanted to keep big monkey patches like this in contrib.

I think we've got different use cases in mind. I'd expect developers to use OC loggers from application code, either by patching logging globally with `use_oc_logging` or using `TraceLogger` for specific logs they want to capture alongside tracing info. It'd be surprising to get the logging integration from the exporter in this case.

OC logging adds trace metadata to regular logs by getting it from the context. Do we still have `traceId` (and etc.) for a given span in context when we export it? By the time it reaches the exporter this is data, not metadata. I'd expect `traceId` to show up in the body of an exporter log, not as a header-like field as it is here.

Reply:
> I think we've got different use cases in mind. I'd expect developers to use OC loggers from application code, either by patching logging globally with `use_oc_logging` or using `TraceLogger` for specific logs they want to capture alongside tracing info. It'd be surprising to get the logging integration from the exporter in this case.

I think we're on the same page. Application developers will have to ask **enrich my logs with trace_id and span_id** explicitly. Exporters will not provide such functionality. The example I gave is that we don't have application developers to explicitly **install** the extension, as we can do it implicitly.
 
> OC logging adds trace metadata to regular logs by getting it from the context. Do we still have `traceId` (and etc.) for a given span in context when we export it?

No, context is unavailable in the exporter code.

> By the time it reaches the exporter this is data, not metadata. I'd expect `traceId` to show up in the body of an exporter log, not as a header-like field as it is here.

Yes, we're on the same page.